cpp/src/sort/is_sorted.cu

```diff
@@ -36,31 +36,27 @@ namespace detail {
 
 auto is_sorted(cudf::table_view const& in,
                std::vector<order> const& column_order,
-               bool has_nulls,
                std::vector<null_order> const& null_precedence,
                rmm::cuda_stream_view stream)
 {
-  // 0-table_view, 1-column_order, 2-null_precedence, 3-validity_columns
-  auto flattened = structs::detail::flatten_nested_columns(in, column_order, null_precedence);
+  auto const comparator =
+    experimental::row::lexicographic::self_comparator{in, column_order, null_precedence, stream};
 
-  auto const d_input           = table_device_view::create(flattened, stream);
-  auto const d_column_order    = make_device_uvector_async(flattened.orders(), stream);
-  auto const d_null_precedence = has_nulls
-                                   ? make_device_uvector_async(flattened.null_orders(), stream)
-                                   : rmm::device_uvector<null_order>(0, stream);
+  if (cudf::detail::has_nested_columns(in)) {
+    auto const device_comparator = comparator.less<true>(has_nested_nulls(in));
```

Comment:
Not sure if it's been discussed before, I feel it makes more sense to move this branching (whether has nested cols or not) into the comparator detail.

Reply:
This is the way it was implemented for `lexicographic::*_comparator`, so I followed the convention. That said, there are some algorithms that dispatch different code-paths for nested and non-nested types already, so you want the granular level of control where those algorithms can select a particular version of the comparator

Reply:
Understood. It's definitely out of the scope of this PR. But if we always dispatch different code-paths, the fine control of those algorithms seems unnecessary.

Reply:
I agree. All of these if-statements scattered throughout the code seems like an anti-pattern we should try to factor out somehow. We should take a look once these are done. I'm also worried about compile time (surprise) and so that would be a reason to consider refactoring before these are complete perhaps.

Reply:
I would prefer any refactor for design patterns to be a follow up to this https://github.com/rapidsai/cudf/issues/12593